lib/omnisharp-atom/features/code-action.ts

```diff
@@ -0,0 +1,67 @@
+import _ = require('lodash');
+import Omni = require('../../omni-sharp-server/omni')
+import ClientManager = require('../../omni-sharp-server/client-manager');
+import OmniSharpAtom = require('../omnisharp-atom')
+import SpacePen = require('atom-space-pen-views');
+import CodeActionsView = require('../views/code-actions-view');
+import Changes = require('./lib/apply-changes');
+
+class CodeCheck {
+
+    private view : SpacePen.SelectListView;
+
+    constructor(private atomSharper: typeof OmniSharpAtom) {
+        this.atomSharper = atomSharper;
+    }
+    public activate() {
+
+        this.atomSharper.addCommand("omnisharp-atom:get-code-actions", () => {
+            ClientManager.getClientForActiveEditor()
+                .subscribe(client => {
+                    client.getcodeactionsPromise(client.makeRequest());
+                });
+
+        });
+
+        Omni.registerConfiguration(client => {
+            client.observeGetcodeactions.subscribe((data) => {
+
+                //pop ui to user.
+                this.view = new CodeActionsView(data.response.CodeActions, () => {
+                    //callback when an item is selected
+                    ClientManager.getClientForActiveEditor()
+                        .subscribe(client => {
+                            client.runcodeactionPromise(client.makeDataRequest<OmniSharp.Models.CodeActionRequest>(
+                                {
+                                    CodeAction : 0, //this is the "ID"(index?) of the code action to run...
```

Comment:
Yes. That's how it is right now.

We're planning on changing it at some point to use the ID of the fix instead.